File: .NET/Examples/WindowsDesktop/Components/IUserControl.cs
using System.Windows.Forms;

interface IUserControl : IContainerControl
{
    // Title of the user control
    string Title { get; }

    // Indicates if the feature shown is enabled
    bool FeatureEnabled { get; }

    // Called periodically in GUI thread
    void PeriodicUpdate();

    // Called when the control is displayed
    void OnOpen();

    // Called when the control is replaced by another
    void OnClose();

}

File: UnderAutomation.UniversalRobots.Showcase.Forms.vb/Components/IUserControl.vb
Imports System.Windows.Forms

Friend Interface IUserControl
    Inherits IContainerControl
    ' Title of the user control
    ReadOnly Property Title As String

    ' Indicates if the feature shown is enabled
    ReadOnly Property FeatureEnabled As Boolean

    ' Called periodically in GUI thread
    Sub PeriodicUpdate()

    ' Called when the control is displayed
    Sub OnOpen()

    ' Called when the control is replaced by another
    Sub OnClose()

End Interface
